### PR TITLE
Additions

### PR DESCRIPTION
**Pull request by e-davidson**
fix bug when only using a custom footer the entire cell was blank.
add pipe or function to transform output
add click event
remove hidden column from column list

### CODE REVIEW

File: projects/table-builder/src/lib/components/column-builder/column-builder.component.ts [hunk 1]
```diff
@@ -32,25 +26,36 @@ export class ColumnBuilderComponent implements OnInit {
   @ViewChild('customCellWrapper') customCellWrapper: TemplateRef<any>;
 
   template: TemplateRef<any>;
+  tranform: (o: any, ...args: any[])=> any ;
 
-  constructor(@Inject(TableBuilderConfigToken) private config: TableBuilderConfig) { }
+  constructor( private tranformCreator: TranformCreator) { }
```
Comment (NachmanBerkowitz) [inline projects/table-builder/src/lib/components/column-builder/column-builder.component.ts:31]: spelling

File: projects/table-builder/src/lib/components/column-builder/column-builder.component.ts [hunk 2]
```diff
@@ -32,25 +26,36 @@ export class ColumnBuilderComponent implements OnInit {
   @ViewChild('customCellWrapper') customCellWrapper: TemplateRef<any>;
 
   template: TemplateRef<any>;
+  tranform: (o: any, ...args: any[])=> any ;
```
Comment (NachmanBerkowitz) [inline projects/table-builder/src/lib/components/column-builder/column-builder.component.ts:29]: spelling

File: projects/table-builder/src/lib/services/tranform-creator.ts [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+import { TableBuilderConfigToken, TableBuilderConfig } from '../classes/TableBuilderConfig';
+
+function isPipe(o : any ): o is PipeTransform {
+  return typeof ((o as PipeTransform).transform ) === 'function';
+}
+
+@Injectable({
+  providedIn:  'root'
+})
+export class TranformCreator{
```
Comment (NachmanBerkowitz) [inline projects/table-builder/src/lib/services/tranform-creator.ts:14]: spelling

**Comment by NachmanBerkowitz** [comment]
Add to header menu.html not to show filter button options if fieldtype is expression

File: projects/table-builder/src/lib/interfaces/report-def.ts [hunk 4]
```diff
@@ -36,6 +38,8 @@ export interface MetaData {
     width?: string;
     noExport?: boolean;
     noFilter?: boolean;
+    tranform?: ( (o: any, ...args: any[])=> any ) | PipeTransform;
```
Comment (NachmanBerkowitz) [inline projects/table-builder/src/lib/interfaces/report-def.ts:41]: spelling

File: projects/table-builder/src/lib/components/column-builder/column-builder.component.ts (hunk 1 above)
Comment (e-davidson) [reply projects/table-builder/src/lib/components/column-builder/column-builder.component.ts]: corrected spelling issue

File: projects/table-builder/src/lib/services/tranform-creator.ts [hunk 5]
```diff
@@ -0,0 +1,40 @@
+import { PipeTransform, Inject, Injectable } from '@angular/core';
```
Comment (NachmanBerkowitz) [inline projects/table-builder/src/lib/services/tranform-creator.ts:1]: there is still a spelling mistake in the file name. Wanna fix it?

**Pull request merged**